drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_rtio_stream.c

```diff
@@ -0,0 +1,357 @@
+		lsm6dsv16x->streaming_sqe = NULL;
+
+		if (data_opt == SENSOR_STREAM_DATA_DROP) {
+
+			/*
+			 * Flush the FIFO by setting the mode to LSM6DSV16X_BYPASS_MODE
+			 *
+			 * STMEMSC API equivalent code:
+			 *
+			 *   lsm6dsv16x_fifo_mode_set(ctx, LSM6DSV16X_BYPASS_MODE);
+			 */
+			struct rtio_sqe *write_fifo_mode = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
+			uint8_t lsm6dsv16x_fifo_mode_set[] = {
+				LSM6DSV16X_FIFO_CTRL4,
+				LSM6DSV16X_BYPASS_MODE,
+			};
+
+			rtio_sqe_prep_tiny_write(write_fifo_mode, iodev,
+						 RTIO_PRIO_NORM, lsm6dsv16x_fifo_mode_set,
+						 ARRAY_SIZE(lsm6dsv16x_fifo_mode_set), NULL);
+
+			rtio_submit(lsm6dsv16x->rtio_ctx, 0);
+		}
+
+		gpio_pin_interrupt_configure_dt(irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		return;
+	}
+
+	uint8_t *buf, *read_buf;
+	uint32_t buf_len, buf_avail;
+	uint32_t req_len = LSM6DSV16X_FIFO_SIZE(fifo_count) + sizeof(struct lsm6dsv16x_fifo_data);
+
+	if (rtio_sqe_rx_buf(lsm6dsv16x->streaming_sqe, req_len, req_len, &buf, &buf_len) != 0) {
+		LOG_ERR("Failed to get buffer");
+		rtio_iodev_sqe_err(lsm6dsv16x->streaming_sqe, -ENOMEM);
+		lsm6dsv16x->streaming_sqe = NULL;
+		gpio_pin_interrupt_configure_dt(irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		return;
+	}
+
+	struct lsm6dsv16x_fifo_data hdr = {
+		.header = {
+			.is_fifo = true,
+			.accel_fs = lsm6dsv16x->accel_fs,
+			.gyro_fs = lsm6dsv16x->gyro_fs,
+			.timestamp = k_ticks_to_ns_floor64(k_uptime_ticks()),
+		},
+		.fifo_count = fifo_count,
+		.accel_batch_odr = lsm6dsv16x->accel_batch_odr,
+		.gyro_batch_odr = lsm6dsv16x->gyro_batch_odr,
+#if defined(CONFIG_LSM6DSV16X_ENABLE_TEMP)
+		.temp_batch_odr = lsm6dsv16x->temp_batch_odr,
+#endif
+	};
+
+	memcpy(buf, &hdr, sizeof(hdr));
+	read_buf = buf + sizeof(hdr);
+	buf_avail = buf_len - sizeof(hdr);
+
+	/*
+	 * Prepare rtio enabled bus to read all fifo_count entries from
+	 * LSM6DSV16X_FIFO_DATA_OUT_TAG.  Then lsm6dsv16x_complete_op_cb
+	 * callback will be invoked.
+	 *
+	 * STMEMSC API equivalent code:
+	 *
+	 *   num = fifo_status.fifo_level;
+	 *
+	 *   while (num--) {
+	 *     lsm6dsv16x_fifo_out_raw_t f_data;
+	 *
+	 *     lsm6dsv16x_fifo_out_raw_get(&dev_ctx, &f_data);
+	 *   }
+	 */
+	struct rtio_sqe *write_fifo_dout_addr = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
+	struct rtio_sqe *read_fifo_dout_reg = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
+	struct rtio_sqe *complete_op = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
+	uint8_t reg = lsm6dsv16x_bus_reg(lsm6dsv16x, LSM6DSV16X_FIFO_DATA_OUT_TAG);
+
+	rtio_sqe_prep_tiny_write(write_fifo_dout_addr, iodev, RTIO_PRIO_NORM, &reg, 1, NULL);
+	write_fifo_dout_addr->flags = RTIO_SQE_TRANSACTION;
+	rtio_sqe_prep_read(read_fifo_dout_reg, iodev, RTIO_PRIO_NORM,
+			   read_buf, buf_avail, lsm6dsv16x->streaming_sqe);
+	read_fifo_dout_reg->flags = RTIO_SQE_CHAINED;
+	if (lsm6dsv16x->bus_type == BUS_I2C) {
+		read_fifo_dout_reg->iodev_flags |= RTIO_IODEV_I2C_STOP | RTIO_IODEV_I2C_RESTART;
+	}
+	rtio_sqe_prep_callback(complete_op,
```

Comment:
probably worth doing...

```c
complete_op->flags |= RTIO_SQE_NO_RESPONSE;
```

Reply:
rtio_sqe_prep_callback_no_cqe() is actually a better answer as it sets the flag in the prep helper